mesonbuild/backend/backends.py

```diff
@@ -429,6 +429,9 @@ def generate_basic_compiler_args(self, target, compiler, no_warn_args=False):
         # NOTE: We must preserve the order in which external deps are
         # specified, so we reverse the list before iterating over it.
         for dep in reversed(target.get_external_deps()):
+            if not dep.found():
+                continue
+
```

Comment:
Maybe we should make `get_compile_args` compiler-aware and make sure we emit empty flags for Vala. I like the idea of having an implicit sanity check for non-found dependencies.